### PR TITLE
[Data format] Serialization for UnitGraph

### PR DESCRIPTION
**Pull request by VoVAllen**
## Description
Serialization for UnitGraph and CSRMatrix

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: src/graph/unit_graph.cc [hunk 1]
```diff
@@ -1152,4 +1152,38 @@ SparseFormat UnitGraph::SelectFormat(SparseFormat preferred_format) const {
...
+  aten::CSRMatrix csr_matrix;
+  CHECK(fs->Read(&csr_matrix)) << "Invalid csr_matrix";
+  SparseFormat restrict_format;
+  CHECK(fs->Read(&restrict_format)) << "Invalid restrict_format";
+  auto mg = CreateUnitGraphMetaGraph(num_vtypes);
+  CSRPtr csr(new CSR(mg, num_src, num_dst, csr_matrix.indptr, csr_matrix.indices, csr_matrix.data));
+  in_csr_ = csr;
+  meta_graph_ = mg;
+  restrict_format_ = restrict_format;
+  // HeteroGraphPtr hptr = CreateFromCSR(num_vtypes, num_src, num_dst, csr_matrix.indptr,
```
Comment (VoVAllen) [inline src/graph/unit_graph.cc]: The code aboves is copied from `CreateFromCSR`. I want to reuse the code in `CreateFromCSR`, but due to my limited C++ skill I don't know how to properly move `hptr` into `this`. Do I need something like a move constructor? Could you give any suggestions? @BarclayII @jermainewang

Comment (BarclayII) [reply src/graph/unit_graph.cc]: I would write `*this = UnitGraph(mg, csr, nullptr, nullptr, restrict_format)`.

**Comment by BarclayII** [comment]
Can you also write a unit test?

**Comment by VoVAllen** [comment]
Sure.

**Comment by VoVAllen** [comment]
@BarclayII I didn't add the support for COO format. The main reason is that I'd like to preserve simplicity in favor of compatibility. Multiple formats need to depend on the implementation detail of the class, which may be subject to change in the future and leads to incompatibility. 
However, I leave the magic number in the serialization, which means we can extend the method in the future based on the magic number.

**Comment by BarclayII** [comment]
Sounds good.

**Pull request merged**